### ISSUE TITLE
Provision Release Manager Quickstarter for every new Project

### ISSUE THREAD

**Issue by metmajer** [opendevstack/ods-provisioning-app#456]
**Is your feature request related to a problem? Please describe.**
The release manager adds overarching capabilities such as component orchestration, parallel build, deploy, test and release, as well as document generation. As projects typically consist of more than one component, projects benefit if the release manager is installed by default into every new project.

**Describe the solution you'd like**
New project -> Release Manager quickstarter is provisioned

**Comment by metmajer** [comment]
FYI @martsec @jorge-romero

**Comment by michaelsauter** [comment]
I would love this to be configurable. I think some ODS installations will benefit from this, while others might not want this (e.g. if you are in a non-documents world, then tying all components together in one pipeline might not be what you want if you believe in the microservices approach)

**Comment by stitakis** [comment]
Currently:
A. the project the creation pipeline needs to finish before running the quickstarter provisioning pipeline. This is because a Jenkins instance needs to be provisioned in `<NEW_PROJECT>-cd` before running the a quickstarter provisioning pipeline in it.
B. ProvApp does not monitor if a creation pipeline failed or completed successfully. 

A solution (that I haven't tested yet) could look like this:
1. To automatically provision the Release Manager quickstarter after project provisioning, ProvApp will need to monitor the project creation pipeline to complete. For this we could add the openshift jar client to ProvApp project to query the oc cluster for the status of the project creation pipeline to trigger then the quickstarter provisioning. 

2. The UI/REST-API could expose a flag that to enable Release Manager quickstarter provision on project creation.

@metmajer @michaelsauter @martsec @jorge-romero does the solution sounds good? Should we explore it?

**Comment by michaelsauter** [comment]
Actually, I just realised that we might run into more hurdles. Right now a project is provisioned with incomplete credentials if a project specific CD user is chosen. Only after this has been resolved manually by the user, quickstarters can be provisioned. Given this situation, I think we'd either need to fix this, or deliver the release manager in a different fashion (as part of the project creation pipeline and not via a separate pipeline).

**Comment by stitakis** [comment]
Maybe a different approach makes the deal and is easier to implement at first step... we could display in the UI a message suggesting the user to provision the Release Manager quickstarter if is missing.

**Comment by clemensutschig** [comment]
This is really tricky to implement - with the old setup and rundeck we could make this fly - as there was a job dependency - and the jobs took a little longer - so you could click create new project and and add a a quickstarter (which in > 90% of all cases worked).. with the jenkins based architecture, and more so - the x2 user ... this will NOT fly anymore ..

**Comment by stitakis** [comment]
@metmajer @michaelsauter @clemensutschig are we doing something related to this in ODS 3 release.

**Comment by michaelsauter** [comment]
I would vote "no" :)

**Comment by metmajer** [comment]
Good enough in 4 for me

**Comment by serverhorror** [comment]
Can we _not_ do that?

We have cases where there is:
* more than one release manager
* no release manager (for good reason)
* a single release manager

**Comment by michaelsauter** [comment]
Aside from @serverhorror's points above, I'd also like to NOT do that because:

* The release manager does come with quite some complexity. It does a lot, and it requires users to learn a lot. I don’t think this is a justified investment for all use cases, e.g. if you do not need the documents. Further, adding the complexity gradually feels better to me than starting with it from day 1.
* I think there are good reasons to deploy component-based. In my eyes, microservices make most sense where services resemble the team structure (e.g. one service = one team). In that world, deployment by team feels most natural to me.

To sum it up, bundling everything up into one big release is not very attractive for me for technical and organizational reasons. If needed for the docs, ok, but personally I see that more of a trade-off then an advantage.

I'm going to go ahead and close this issue. Please re-open if you strongly disagree @metmajer.
